src/pages/settings/Profile/PersonalDetails/AddressPage.js

```diff
@@ -0,0 +1,226 @@
+import CONST from '../../../../CONST';
+import TextInput from '../../../../components/TextInput';
+import styles from '../../../../styles/styles';
+import Navigation from '../../../../libs/Navigation/Navigation';
+import * as PersonalDetails from '../../../../libs/actions/PersonalDetails';
+import compose from '../../../../libs/compose';
+import AddressSearch from '../../../../components/AddressSearch';
+import CountryPicker from '../../../../components/CountryPicker';
+import StatePicker from '../../../../components/StatePicker';
+
+const propTypes = {
+    /* Onyx Props */
+
+    /** User's private personal details */
+    privatePersonalDetails: PropTypes.shape({
+        /** User's home address */
+        address: PropTypes.shape({
+            street: PropTypes.string,
+            city: PropTypes.string,
+            state: PropTypes.string,
+            zip: PropTypes.string,
+            country: PropTypes.string,
+        }),
+    }),
+
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    privatePersonalDetails: {
+        address: {
+            street: '',
+            city: '',
+            state: '',
+            zip: '',
+            country: '',
+        },
+    },
+};
+
+class AddressPage extends Component {
+    constructor(props) {
+        super(props);
+
+        this.validate = this.validate.bind(this);
+        this.updateAddress = this.updateAddress.bind(this);
+        this.onCountryUpdate = this.onCountryUpdate.bind(this);
+
+        const currentCountry = lodashGet(props.privatePersonalDetails, 'address.country') || '';
+        this.state = {
+            isUsaForm: currentCountry === CONST.USA_COUNTRY_NAME,
+        };
+    }
+
+    /**
+     * @param {String} newCountry - new country selected in form
+     */
+    onCountryUpdate(newCountry) {
+        if (newCountry === CONST.USA_COUNTRY_NAME) {
+            this.setState({isUsaForm: true});
+        } else {
+            this.setState({isUsaForm: false});
+        }
+    }
+
+    /**
+     * Submit form to update user's first and last legal name
+     * @param {Object} values - form input values
+     */
+    updateAddress(values) {
+        PersonalDetails.updateAddress(
+            values.addressLine1.trim(),
+            values.addressLine2.trim(),
+            values.city.trim(),
+            values.state.trim(),
+            values.zipPostCode,
+            values.country,
+        );
+    }
+
+    /**
+     * @param {Object} values - form input values
+     * @returns {Object} - An object containing the errors for each inputID
+     */
+    validate(values) {
+        const errors = {};
+
+        const requiredFields = [
+            'addressLine1',
+            'city',
+            'zipPostCode',
+            'country',
+            'state',
+        ];
+
+        // Check "State" dropdown is a valid state if selected Country is USA.
+        if (this.state.isUsaForm && !COMMON_CONST.STATES[values.state]) {
```

Comment:
[This issue](https://github.com/Expensify/App/issues/21793) was not handled in this PR.

Issue: Error is shown in state/province field of Home address, Personal details while changing to a different location than US.